src/components/More.js

```diff
@@ -33,10 +34,29 @@ export class More extends React.Component {
     super(props);
     this.state = {expanded: false};
   }
+
+  // https://discuss.reactjs.org/t/how-to-calculate-jqyery-closest-in-react/1988
+  componentDidMount() {
+    let app = document.getElementById('app');
```

Comment:
This is fragile since you are relying on an element with id='app' being on the page.

It would be better to either
a) attach the click handler to document.body
... or ...
b) render a clickable backdrop with css like `{position: "absolute", top: 0, left: 0, right: 0, bottom: 0}`